nominal/core.py

```diff
@@ -755,6 +917,94 @@ def _search_datasets(self) -> Iterable[Dataset]:
         response = self._catalog_client.search_datasets(self._auth_header, request)
         for ds in response.results:
             yield Dataset._from_conjure(self, ds)
+        
+    def _get_user_rid_from_email(self, user_email: str) -> str:
+        request = authentication_api.SearchUsersRequest(
+            query=authentication_api.SearchUsersQuery(
+                exact_match=user_email,
+            )
+        )
+        response = self._authentication_client.search_users_v2(
+            self._auth_header,
+            request
+        )
+        if len(response.results) == 0:
+            raise ValueError(f"User {user_email} not found")
+        if len(response.results) > 1:
+            raise ValueError(f"Found multiple users with email {user_email}")
+        return response.results[0].rid
```

Comment:
eventually we may want to change the checklists api to take in a user email but I think we're just gonna do this to unblock for now